vsintegration/src/FSharp.Editor/Common/Pervasive.fs

```diff
@@ -200,4 +200,27 @@ module Async =
                     let! replyCh = agent.Receive ()
                     replyCh.Reply res 
             }
-        async { return! agent.PostAndAsyncReply id }
\ No newline at end of file
+        async { return! agent.PostAndAsyncReply id }
+
+let taskMap f t =
+    backgroundTask {
+        let! result = t
+        return f result
+    }
+
+let rec private interleave' result seqs =
+    match seqs with
+    | [] -> result
+    | _ ->
+        let items, newSeqs =
+            [ for s in seqs do
+                match s |> Seq.tryHead with
+                | Some item -> item, Seq.tail s
+                | None -> () ]
+            |> List.unzip
+
+        interleave' (Seq.append result items) newSeqs
+
+/// Combines sequences of varying lengths into one by interleaving the elements
+/// E.g. aaaaa bb ccc -> abcabcacaa
+let interleave seqs = seqs |> Seq.toList |> interleave' Seq.empty
```

Comment:
TBH I am not convinced that interleaving should work better than standard flattening.
Data from different documents of the same project is more likely to be collocated in memory and follow similar cache hits.

Why the decision to interleave instead of flatten?

Reply:
The idea is that files from the same projects are dependent on each other and could spend time just waiting. Whereas for different projects work can continue independently (this is how it was previously, projects processed in parallel, but individual files sequentially). 

But it could very well be slower that way, so we should figure out a way to measure it. Which should be possible with a synthetic benchmark.

Also might experiment with sorting the documents one way or another.